difflint/lib.py

```diff
@@ -2,16 +2,49 @@
 from datetime import datetime
 from difflib import unified_diff
 from io import StringIO
-from os import unlink
+from os import devnull, getcwd, unlink
+import os.path
 import re
-from subprocess import call, check_output
+from subprocess import call, CalledProcessError, check_output
 from sys import stderr
 
 from .lint import get_missing_configuration_files, get_missing_linters, lint
 
 LOG_FILE = 'lintdiff.log'
 MISSING_FILE_EXIT_CODE = 72  # os.EX_OSFILE is not portable
 
+def make_path_from_project_root(*args):
+    """
+    Builds a string containing the project root as the prefix and appends
+    zero or more strings to the end representing pieces of a filepath.
+
+    Inputs:
+        *args: Zero or more strings comprising the suffix of a filepath,
+               excluding seperators such as forward or backslashes.
+
+    Output: A platform-aware filepath string with the project root
+            directory as the prefix and any suffix arugments at the end.
+
+    Example:
+        (Run from the directory ~/my_repo/party_hard/beta/ on a Linux system.)
+
+        > make_path_from_project_root("math", "is", "fun.txt")
+        > /Users/devinekins/my_repo/math/is/fun.txt
+    """
+    directory = getcwd()
+
+    # While we aren't looking at the filesystem root, find the folder
+    # containing a .git folder.
+    while directory != os.path.abspath(os.sep):
```

Comment:
See also https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.parents, which will let you just loop over a file's ancestors directly.